backend/controllers/task.controller.js

```diff
@@ -0,0 +1,100 @@
+const { Task } = require('../models')
+
+exports.createTask = async (req, res) => {
+    const { title, description, type, deadline } = req.body
+    const user = req.user
+    try {
+        console.log(user)
+        let task = {
+            title: title,
+            description: description,
+            type: type,
+            deadline: deadline,
+            user: user,
+        }
+
+        let row = new Task(task)
+
+        await row.save()
+
+        return res.json({ msg: 'Task Created Successfully' })
+    } catch (err) {
+        console.log(err)
+        return res.json({ err: err })
+    }
+}
+
+exports.updateTask = async (req, res) => {
+    const { title, description, type, deadline } = req.body
+    const user = req.user
+    try {
+        let task = {
+            title: title,
+            description: description,
+            type: type,
+            deadline: deadline,
+            user: user._id,
+        }
+
+        let row = await Task.findByIdAndUpdate(req.params.id, task)
+
+        return res.json({ msg: 'Task Updated Successfully' })
+    } catch (err) {
+        return res.json({ err: err })
+    }
+}
+
+exports.deleteTask = async (req, res) => {
+    try {
+        await Task.findByIdAndDelete(req.params.id)
+
+        return res.json({ msg: 'Task Deleted Successfully' })
+    } catch (err) {
+        return res.json({ err: err })
+    }
+}
+
+exports.getTask = async (req, res) => {
+    try {
+        let row = await Task.findById(req.params.id)
+
+        return res.json(row)
+    } catch (err) {
+        return res.json({ err: err })
+    }
+}
+
+exports.getTasks = async (req, res) => {
+    try {
+        console.log(req.user)
+        let rows = await Task.find({ user: req.user._id })
+
+        return res.json(rows)
+    } catch (err) {
+        console.log(err)
+        return res.json({ err: err })
+    }
+}
+
+exports.updateTaskStatus = async (req, res) => {
+    try {
+        let task = await Task.findById(req.params.id)
+        user = req.user
+        task.status = !task.status
+
+        if (task.type in ['personal', 'professional']) {
+            user.points += task.status ? 5 : -5
```

Comment:
why is this -5 & -10 given, user points should never decrease

Reply:
Same as in subtask controller, If the user unmark the task, then the points will be reversed/deducted